### DISCUSSION TITLE
Loose prisma datasource db provider.

### DISCUSSION THREAD

**Discussion by muazhari** [langfuse/langfuse#1461] [Ideas]
### Describe the feature or potential improvement

I currently using cockroachdb as the main database. It is a waste if I spin up postgresql only for langfuse. When I change the `DATABASE_URL` to `cockroachdb://${USER}@${HOST}:${PORT}/${DATABASE}`, this error came up:
```
Error: P1013: The provided database string is invalid. The scheme is not recognized in the database URL. Please refer to the documentation in https://www.prisma.io/docs/reference/database-reference/connection-urls for constructing a correct connection string. In some cases, certain characters must be escaped. Please check the string for any illegal characters.
```

I suggest to change the code in [`./web/prisma/schema.prisma`](https://github.com/langfuse/langfuse/blob/44589f33b21a6d52e64d526465d7fab95716d36e/web/prisma/schema.prisma#L10) from:
```
datasource db {
    provider          = "postgresql"
    url               = env("DATABASE_URL")
    directUrl         = env("DIRECT_URL")
    shadowDatabaseUrl = env("SHADOW_DATABASE_URL")
}
```
to like this:
```
datasource db {
    provider          = env("DATABASE_PROVIDER")
    url               = env("DATABASE_URL")
    directUrl         = env("DIRECT_URL")
    shadowDatabaseUrl = env("SHADOW_DATABASE_URL")
}
```

**Comment by marcklingen** [comment]
Hi @muazhari,

we currently run all CI for postgres and also include some custom sql that is postgres specific. Thus, this will not work and adds a lot of additional complexity across langfuse to add.

Is it difficult for you to spin up a new db or just run it locally via docker?
